perf/impl/https/v0.1/main.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/binary"
+	"encoding/json"
+	"encoding/pem"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"math/big"
+	"net"
+	"net/http"
+	"time"
+)
+
+const (
+	BlockSize = 64 << 10
+)
+
+func handleRequest(w http.ResponseWriter, r *http.Request) {
+	// Read the big-endian bytesToSend value
+	var bytesToSend uint64
+	if err := binary.Read(r.Body, binary.BigEndian, &bytesToSend); err != nil {
+		http.Error(w, "failed to read uint64 value", http.StatusBadRequest)
+		return
+	}
+
+	// Read and discard the remaining bytes in the request
+	io.Copy(io.Discard, r.Body)
+
+	buf := make([]byte, BlockSize)
+
+	for bytesToSend > 0 {
```

Comment:
Please set the `content-type` and `content-length` headers and then call `WriteHeader(OK)` such that the Go stdlib doesn't do extra work here (https://cs.opensource.google/go/go/+/refs/tags/go1.20.4:src/net/http/server.go;drc=ea4631cc0cf301c824bd665a7980c13289ab5c9d;l=120).

Reply:
Agree, setting the content length seems reasonable.

Reply:
Good idea. Thanks @MarcoPolo. Done for the client and server in c0c34f4.